### PR TITLE
[datadog_dashboard] Add run-workflow widget definition

### PR DESCRIPTION
**Pull request by DavidRobertAnsart**
## Changes

- Add run workflow widget definition to dashboard resource.
- Add test and recorded cassettes for the new widget.
- Add doc.

## How to test:

1. Build the provider locally.
2. Try this dashboard resource:
```tf
resource "datadog_dashboard" "ordered_dashboard" {
  title       = "Run workflow terraform dashboard"
  layout_type = "ordered"
  template_variable {
    name             = "Env"
    prefix           = "Env"
    defaults         = ["*"]
    available_values = ["prod", "staging", "test"]
  }
  widget {
    run_workflow_definition {
      title       = "My workflow widget"
      title_size  = "13"
      title_align = "left"
      workflow_id = "2e055f16-8b6a-4cdd-b452-17a34c44b160"
      input {
        name  = "env"
        value = "$Env.value"
      }
    }
    widget_layout {
      x      = 0
      y      = 0
      width  = 4
      height = 4
    }
  }
  notify_list = []
  reflow_type = "fixed"
}
```

### CODE REVIEW

**Review by nkzou — DISMISSED** [review]

File: docs/resources/dashboard.md [hunk 1]
```diff
@@ -11288,6 +11327,43 @@ Optional:
...
+
+<a id="nestedblock--widget--run_workflow_definition--custom_link"></a>
+### Nested Schema for `widget.run_workflow_definition.custom_link`
+
+Optional:
+
+- `is_hidden` (Boolean) The flag for toggling context menu link visibility.
+- `label` (String) The label for the custom link URL.
+- `link` (String) The URL of the custom link.
+- `override_label` (String) The label id that refers to a context menu link item. When override_label is provided, the client request omits the label field.
```
Comment (cswatt) [inline docs/resources/dashboard.md:11354]:
```suggestion
- `override_label` (String) The label ID that refers to a context menu link item. When `override_label` is provided, the client request omits the label field.
```

File: docs/resources/dashboard.md [hunk 2]
```diff
@@ -11288,6 +11327,43 @@ Optional:
...
+
+
+<a id="nestedblock--widget--run_workflow_definition--input"></a>
+### Nested Schema for `widget.run_workflow_definition.input`
+
+Required:
+
+- `name` (String) Name of the workflow input.
+- `value` (String) Dashboard template variable. Can be suffixed with '.value' or '.key'.
```
Comment (cswatt) [inline docs/resources/dashboard.md:11363]:
```suggestion
- `value` (String) Dashboard template variable. Can be suffixed with `.value` or `.key`.
```

File: docs/resources/dashboard.md [hunk 3]
```diff
@@ -11288,6 +11327,43 @@ Optional:
 
 
 
+<a id="nestedblock--widget--run_workflow_definition"></a>
+### Nested Schema for `widget.run_workflow_definition`
+
+Required:
+
+- `workflow_id` (String) Workflow id
```
Comment (cswatt) [inline docs/resources/dashboard.md:11335]:
```suggestion
- `workflow_id` (String) Workflow ID
```

File: docs/resources/dashboard.md [hunk 4]
```diff
@@ -5814,6 +5816,43 @@ Optional:
 
 
 
+<a id="nestedblock--widget--group_definition--widget--run_workflow_definition"></a>
+### Nested Schema for `widget.group_definition.widget.run_workflow_definition`
+
+Required:
+
+- `workflow_id` (String) Workflow id
```
Comment (cswatt) [inline docs/resources/dashboard.md:5824]:
```suggestion
- `workflow_id` (String) Workflow ID
```

File: docs/resources/dashboard.md [hunk 5]
```diff
@@ -5814,6 +5816,43 @@ Optional:
...
+
+<a id="nestedblock--widget--group_definition--widget--run_workflow_definition--custom_link"></a>
+### Nested Schema for `widget.group_definition.widget.run_workflow_definition.custom_link`
+
+Optional:
+
+- `is_hidden` (Boolean) The flag for toggling context menu link visibility.
+- `label` (String) The label for the custom link URL.
+- `link` (String) The URL of the custom link.
+- `override_label` (String) The label id that refers to a context menu link item. When override_label is provided, the client request omits the label field.
```
Comment (cswatt) [inline docs/resources/dashboard.md:5843]:
```suggestion
- `override_label` (String) The label ID that refers to a context menu link item. When `override_label` is provided, the client request omits the label field.
```

File: docs/resources/dashboard.md [hunk 6]
```diff
@@ -5814,6 +5816,43 @@ Optional:
...
+
+
+<a id="nestedblock--widget--group_definition--widget--run_workflow_definition--input"></a>
+### Nested Schema for `widget.group_definition.widget.run_workflow_definition.input`
+
+Required:
+
+- `name` (String) Name of the workflow input.
+- `value` (String) Dashboard template variable. Can be suffixed with '.value' or '.key'.
```
Comment (cswatt) [inline docs/resources/dashboard.md:5852]:
```suggestion
- `value` (String) Dashboard template variable. Can be suffixed with `.value` or `.key`.
```

**Review by cswatt — CHANGES_REQUESTED** [review]
small nitpicky things

**Review by skarimo — APPROVED** [review]

**Review by DrkSephy — APPROVED** [review]
Thanks for this contribution 🙌

**Review by cswatt — APPROVED** [review]

**Pull request merged**